packages/hash/frontend/src/blocks/page/ComponentView.tsx

```diff
@@ -196,20 +204,127 @@ export class ComponentView implements NodeView<Schema> {
     }
   }
 
-  editableRef = (editableNode: HTMLElement) => {
-    if (
-      this.contentDOM &&
-      editableNode &&
-      !editableNode.contains(this.contentDOM)
-    ) {
-      editableNode.appendChild(this.contentDOM);
-      this.contentDOM.style.display = "";
+  private getDraftBlockEntity() {
+    const draftId = this.getBlockDraftId();
+    const entity = this.store.draft[draftId];
+
+    if (!entity || !isDraftBlockEntity(entity)) {
+      throw new Error("Component view can't find block entity");
+    }
+
+    return entity;
+  }
+
+  private getBlockDraftId() {
+    return this.editorView.state.doc.resolve(this.getPos()).node(2).attrs
+      .draftId;
+  }
+
+  private onBlockLoaded = () => {
+    /**
+     * After two calls to setImmediate, we know the block will have had the
+     * chance to initiate an editable section and thereby become selected if
+     * initiated via the suggester. If it hasn't happened in that time, we want
+     * to expire the opportunity to become automatically selected.
+     *
+     * @todo find a better way of knowing there's been the opportunity for
+     *       this which doesn't depend on timing
+     */
+    if (this.wasSuggested) {
+      setImmediate(() =>
+        setImmediate(() => {
+          this.wasSuggested = false;
+        }),
+      );
+    }
+  };
+
+  private editableRef = (editableNode: HTMLElement | null) => {
+    const childTextEntity = getChildDraftEntityFromTextBlock(
+      this.getBlockDraftId(),
+      this.store,
+    );
+
+    if (!childTextEntity) {
+      throw new Error("Block not ready to become editable");
+    }
+
+    const state = this.editorView.state;
+    let tr: Transaction<Schema> | null = null;
+
+    if (!isTextEntity(childTextEntity)) {
+      tr ??= state.tr;
+
+      const newTextDraftId = generateDraftIdForEntity(null);
+      addEntityStoreAction(state, tr, {
+        type: "newDraftEntity",
+        payload: {
+          accountId: childTextEntity.accountId,
+          draftId: newTextDraftId,
+          entityId: null,
+        },
+      });
+
+      // @todo should we use the text entity directly, or just copy the content?
+      addEntityStoreAction(state, tr, {
+        type: "updateEntityProperties",
+        payload: {
+          draftId: newTextDraftId,
+          // @todo indicate the entity type?
+          properties: textBlockNodeToEntityProperties(this.node),
+          merge: false,
+        },
+      });
+
+      addEntityStoreAction(state, tr, {
+        type: "updateEntityProperties",
+        payload: {
+          draftId: childTextEntity.draftId,
+          properties: {
+            text: {
+              __linkedData: {},
+              data: entityStorePluginStateFromTransaction(tr, state).store
+                .draft[newTextDraftId]!,
+            },
+          },
+          merge: true,
+        },
+      });
```

Comment:
This is just moved from the block creation point to the point at which the block becomes editable